### DISCUSSION TITLE
unable to update IPV6

### DISCUSSION THREAD

**Discussion by Narflar** [qdm12/ddns-updater#279] [Q&A] [answered]
Hello, I'm having an issue with IPV6 updates.  It doesn't seem that the IPV6 address is being grabbed.  Below are my logs, IPV4 is working fine.  When I goto https://api64.ipify.org/ I get an ipv6 address.

DEBUG configured to fetch IP: v4 or v6: false, v4: true, v6: true
WARN obtaining ipv6 address: try 1 of 3: no IP address found: from "https://api64.ipify.org/" for version ipv6
WARN obtaining ipv6 address: try 2 of 3: no IP address found: from "https://api64.ipify.org/" for version ipv6
WARN obtaining ipv6 address: try 3 of 3: no IP address found: from "https://api64.ipify.org/" for version ipv6
DEBUG your public IP address are: v4 or v6: <nil>, v4: 0.0.0.0, v6: <nil>
ERROR no IP address found: from "https://api64.ipify.org/" for version ipv6

I changed the environment variables to do all http providers and DNS providers instead of just https://api64.ipify.org and LOG_CALLER to short instead of hidden. Below is what I'm getting in my log for ipv6...

run.go:268: DEBUG configured to fetch IP: v4 or v6: false, v4: true, v6: true
getip.go:21: WARN obtaining ipv6 address: try 1 of 3: dial udp6 [2606:4700:4700::1111]:53: connect: cannot assign requested address
getip.go:21: WARN obtaining ipv6 address: try 2 of 3: Get "http://ip1.dynupdate6.no-ip.com": dial tcp [2a07:dc00:1330:2::204]:80: connect: cannot assign requested address
getip.go:21: WARN obtaining ipv6 address: try 3 of 3: dial udp6 [2001:4860:4802:32::a]:53: connect: cannot assign requested address
run.go:270: DEBUG your public IP address are: v4 or v6: , v4: 0.0.0.0, v6:
run.go:272: ERROR dial udp6 [2001:4860:4802:32::a]:53: connect: cannot assign requested address
run.go:223: DEBUG IPv4 address of something.com is 0.0.0.0 and your IPv4 address is 0.0.0.0, skipping update
run.go:231: DEBUG IPv6 address of something.com is and your IPv6 address is , skipping update

**Comment by qdm12** [comment, marked as answer]
I don't think https://api64.ipify.org/ is built-in the program right? It also doesn't look IPv6 exclusive so it won't work if you have both an ipv4 and ipv6 address.

Now the other error messages shows your docker isn't configured to support IPv6. Try playing around with

```sh
docker run --rm alpine:3.15 wget -O- http://ip1.dynupdate6.no-ip.com
```

and adjust your docker daemon settings / container flags until you get your ipv6 address and no error.

**Reply by Narflar** [reply]
thanks, the address was a typo.  It should have been https://api6.ipify.org/.  either way, you hit the nail on the head about docker not being configured for ipv6.  seems docker doesn't have ipv6 fully fleshed out and setting it up isn't worth the effort for what I'm doing.  Thank you for pointing me in the right direction.

**Reply by qdm12** [reply]
I'll work on making it able to run outside Docker as a statically built binary. There is no reason for it to run inside Docker really.
And attach pre-built binaries to the Github release. That should work better with IPv6. You can subscribe to https://github.com/qdm12/ddns-updater/issues/294 to be notified when it's done.

**Reply by felixwrt** [reply]
I'm using `docker-compose` and also ran into this issue (`cannot assign requested address`) when trying to update ipv6 records. I didn't want to enable ipv6 support in Docker as it's still experimental, but I found another solution: You can specify `network_mode: host` in the `docker-compose.yml` file, which lets the container use the host network. This way, the container can use ipv6. You're loosing network isolation this way, but for this service, I don't mind.

**Reply by qdm12** [reply]
FYI everyone there are now static binaries for each release, so you could also use that. Thanks!
